_posts/2016-11-29-future.md

```diff
@@ -0,0 +1,213 @@
+
+Check out the [6.16.0 blog post](https://babeljs.io/blog/2016/09/28/6.16.0#new-feature) for more info!
+
+### Code Coverage / Instrumentation
+
+With tools like [nyc](https://github.com/istanbuljs/nyc) and [babel-plugin-instanbul](https://github.com/istanbuljs/babel-plugin-istanbul).
+
+## Future
+
+### [Plugin API Changes](https://github.com/babel/notes/blob/master/2016-08/august-01.md#potential-api-changes-for-traversal)
+
+There is a lot of confusion around how plugins/presets interact regarding ordering. This results in bugs and issues with the config which require users to place plugins before/after others in a non-intuitive way.
+
+This is a fundamental change to the core of Babel which will take some work to figure out.
+
+### Typescript support
+
+Be able to understand typescript syntax (like we do with flow), add a plugin to strip typescript types for better interop.
+
+### Use Type Information
+
+Integrate with a type system such as flow/typescript to make optimizations.
+
+### Take in a Dependency Graph
+
+Integrate with other tools like webpack/bundlers? This will allow cross file transformations or whole codebase optimizations. The main reason for this would be for the minifier (being able to remove properties based on checking the usage across the whole application) or providing errors for missing/invalid import/exports.
+
+### Parser Errors
+
+Better error messages from the parser like in [Compiler Errors for Humans](http://elm-lang.org/blog/compiler-errors-for-humans).
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">babel-eslint@7.1.1: now adds the code frame when there&#39;s a parser error! <a href="https://t.co/yoxRpGXq5E">pic.twitter.com/yoxRpGXq5E</a></p>&mdash; Henry Zhu (@left_pad) <a href="https://twitter.com/left_pad/status/799388723896946692">November 17, 2016</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+> It is obvious that we all want to see helpful errors!
+
+- [babel/babylon#125](https://github.com/babel/babylon/pull/125): Instead of "Unexpected token" explain that `await` needs to be used in an async function.
+- Do better inference/guessing on user intention to prevent vague errors.
+
+### `babel-init`
+
+Basically a way to setup babel easier like create-react-app does.
+
+- Setup a .babelrc from scratch
+- Maybe reuse `babel` npm package to default to use babel-preset-env
+
+### Run [tc39/test262](https://github.com/tc39/test262)
+
+Running official spec tests against Babel to make sure we comply to the spec or at least know when we don't.
+
+### Smoke Tests
+
+It would be useful to either have a reverse [greenkeeper](https://greenkeeper.io) or to run tests with master branch of Babel so that we can catch any major regressions before any release. In theory we would want to take the biggest open source projects that use babel and run their tests.
```

Comment:
Yeah I've wanted that before and why we tried to implement canary in lerna for Babel. However we haven't tired it due to not being confident about what happens with multiple dependencies and transitive deps given Babel is composed of other packages and other projects are just 1